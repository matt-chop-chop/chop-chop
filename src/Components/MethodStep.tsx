import { Checkbox, Flex, Text } from "@chakra-ui/react";

type MethodStepProps = {
  step: string;
  checked: boolean;
  setChecked: (checked: boolean) => void;
};

const MethodStep = ({ step, checked, setChecked }: MethodStepProps) => {
  return (
    <Flex
      background="var(--chakra-colors-light-background)"
      border="2px solid var(--chakra-colors-light-text)"
      borderRadius={8}
      mt={4}
      onClick={(e) => {
        e.preventDefault();
        setChecked(!checked);
      }}
      p={4}
      transition="transform 250ms ease-out"
      _hover={{
        cursor: "pointer",
        transition: "transform 250ms ease-in",
        transform: "scale(1.01)",
      }}
    >
      <Checkbox
        isChecked={checked}
        mr={4}
        onClick={() => {
          setChecked(!checked);
        }}
      />
      <Text variant="body-large">{step}</Text>
    </Flex>
  );
};

export default MethodStep;
