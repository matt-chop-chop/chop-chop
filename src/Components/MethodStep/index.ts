export { default as MethodStep } from "./MethodStep";
