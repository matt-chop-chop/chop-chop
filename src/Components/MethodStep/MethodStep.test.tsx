import { shallow } from "enzyme";
import React from "react";
import MethodStep, { MethodStepProps } from "./MethodStep";

const setChecked = jest.fn();

const props: MethodStepProps = {
  step: "This is a test Method Step",
  setChecked: setChecked,
  checked: false,
};

describe("MethodStep", () => {
  it("is exported from MethodStep", () => {
    expect(MethodStep).toBeDefined();
  });

  it("renders as expected ", () => {
    const wrapper = shallow(<MethodStep {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("calls setChecked when the Checkbox is clicked", () => {
    const wrapper = shallow(<MethodStep {...props} />);

    const Checkbox = wrapper.find("Checkbox");
    expect(Checkbox).toHaveLength(1);

    Checkbox.simulate("click");

    expect(setChecked).toHaveBeenCalledWith(true);
  });

  it("calls setChecked when the Flex box is clicked", () => {
    setChecked.mockClear();
    const wrapper = shallow(<MethodStep {...props} />);

    const Checkbox = wrapper.find("Flex");

    Checkbox.simulate("click", {
      preventDefault: () => {},
    });

    expect(setChecked).toHaveBeenCalledWith(true);
  });
});
