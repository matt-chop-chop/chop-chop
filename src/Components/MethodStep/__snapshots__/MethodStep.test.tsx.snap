// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MethodStep renders as expected  1`] = `
<Flex
  _hover={
    {
      "cursor": "pointer",
      "transform": "scale(1.01)",
      "transition": "transform 250ms ease-in",
    }
  }
  background="var(--chakra-colors-light-background)"
  border="2px solid var(--chakra-colors-light-text)"
  borderRadius={8}
  mt={4}
  onClick={[Function]}
  p={4}
  transition="transform 250ms ease-out"
>
  <Checkbox
    isChecked={false}
    mr={4}
    onClick={[Function]}
  />
  <Text
    variant="body-large"
  >
    This is a test Method Step
  </Text>
</Flex>
`;
